app/optscale_api/auth_api.py

```diff
@@ -2,95 +2,74 @@
 
-        except Exception as error:
-            logger.error(f"Error obtaining auth token with user credentials: {error}")
-        return None
+class OptScaleAuth:
+    def __init__(self):
+        self.api_client = APIClient(base_url=settings.opt_scale_api_url)
 
-    async def obtain_user_auth_token_with_admin_api_key(self, user_id: str,
-                                                        admin_api_key: str) -> str | None:
+    async def obtain_user_auth_token_with_admin_api_key(
+        self, user_id: str, admin_api_key: str
+    ) -> str | Exception:
         """
         Obtains an authentication token for the given user_id using the admin API key
         :param user_id: the user's ID for whom the access token will be generated
         :type user_id: string
         :param admin_api_key: the secret API key
         :type admin_api_key: string
-        :return: The user authentication token if successfully obtained and verified, o
-        therwise None.
+        :return: The user authentication token if successfully obtained and verified,
+        otherwise a UserAccessTokenError exception
 
         """
         payload = {"user_id": user_id}
-        headers = self.build_admin_api_key_header(admin_api_key=admin_api_key)
-        try:
-            response = await self.api_client.post(endpoint=AUTH_TOKEN_ENDPOINT,
-                                                  headers=headers,
-                                                  data=payload
-                                                  )
-            if not response:
-                logger.error("Empty response when getting the user auth token with admin API key.")
-                return None
-            if response.get("user_id") != user_id:
-                logger.error(f"User ID mismatch: requested {user_id}, "
-                             f"received {response.get('user_id')}")
-                return None
-            token = response.get("token")
-            if token is not None:
-                return token
-            logger.warning("Token not found in response when getting user auth token.")
-
-        except Exception as error:
-            logger.error(f"Error obtaining user auth token with admin API key: {error}")
+        headers = build_admin_api_key_header(admin_api_key=admin_api_key)
+        response = await self.api_client.post(
+            endpoint=AUTH_TOKEN_ENDPOINT, headers=headers, data=payload
+        )
+        if response.get("error"):
+            logger.error(f"Failed to get an admin access token for user {user_id}")
+            raise OptScaleAPIResponseError(
+                title="Error response from OptScale",
+                reason=response.get("data", {}).get("error", {}).get("reason", ""),
+                status_code=response.get("status_code", http_status.HTTP_403_FORBIDDEN),
+            )
 
-        return None
+        if response.get("data", {}).get("user_id", 0) != user_id:
```

Comment:
Is 0 a special value? Because otherwise this may cause a bug if the first user's ID is 0

Reply:
no it's not a special value. the used ID can never be 0. It's just to have the default value as a int that will be always different from the incoming ID